src/libYARP_OS/src/Node.cpp

```diff
@@ -12,19 +12,50 @@
 #include <yarp/os/impl/Logger.h>
 #include <yarp/os/NestedContact.h>
 #include <yarp/os/Port.h>
+#include <yarp/os/PortReport.h>
+#include <yarp/os/PortInfo.h>
 #include <yarp/os/Network.h>
 #include <yarp/os/RosNameSpace.h>
 #include <yarp/os/impl/PlatformStdlib.h>
 #include <yarp/os/impl/NameClient.h>
 
+#include <algorithm>
+#include <vector>
 #include <list>
 #include <map>
 
 using namespace yarp::os;
 using namespace yarp::os::impl;
 
-static ConstString toRosName(const ConstString& str)
-{
+class ROSReport : public PortReport {
+public:
+    std::multimap<std::string, std::string> outgoingURIs;
+    std::multimap<std::string, std::string> incomingURIs;
+
+    ROSReport() {
+    }
+
+    virtual void report(const PortInfo& info) {
+        if (info.tag == PortInfo::PORTINFO_CONNECTION) {
+            NameClient& nic = NameClient::getNameClient();
+            Contact c;
+            if (info.incoming) {
+                c = RosNameSpace::rosify(nic.queryName(info.sourceName));
+
+            } else {
+                c = RosNameSpace::rosify(nic.queryName(info.targetName));
+            }
+
+            if (info.incoming) {
+                incomingURIs.insert(std::make_pair(info.portName, c.toURI()));
+            } else {
+                outgoingURIs.insert(std::make_pair(info.portName, c.toURI()));
+            }
+        }
```

Comment:
@drdanz I think it's nicer to merge the two if-else constructs into one. Are you happy with that? Will do in this case.
```c++
            if (info.incoming) {
                c = RosNameSpace::rosify(nic.queryName(info.sourceName));
                incomingURIs.insert(std::make_pair(info.portName, c.toURI()));
            } else {
                c = RosNameSpace::rosify(nic.queryName(info.targetName));
                outgoingURIs.insert(std::make_pair(info.portName, c.toURI()));
            }
```